lib/resty/kafka/producer.lua

```diff
@@ -301,14 +323,29 @@ _timer_flush = function (premature, self, time)
 end
 
 
+local function shuffle_table(t)
+    local rand = math.random
+    local iterations = #t
+    local j
+
+    for i = iterations, 2, -1 do
+        j = rand(i)
+        t[i], t[j] = t[j], t[i]
+    end
+end
+
 function _M.new(self, broker_list, producer_config, cluster_name)
+    math.randomseed( ngx.time() )
+
     local name = cluster_name or DEFAULT_CLUSTER_NAME
     local opts = producer_config or {}
     local async = opts.producer_type == "async"
     if async and cluster_inited[name] then
         return cluster_inited[name]
     end
 
+    shuffle_table(broker_list)
```

Comment:
May be you can `shuffle_table` outside `resty.kakfa`, this method seems not must in kafka protocol?